src/rHelpAliases.ts

```diff
@@ -0,0 +1,114 @@
+    private allPackageAliases?: {
+        [key: string]: PackageAliases;
+    }
+    private aliases?: rHelpPanel.Alias[];
+
+    constructor(args: rHelpPanel.AliasProviderArgs){
+        this.rPath = args.rPath;
+        this.rScriptFile = args.rScriptFile;
+    }
+
+    // delete stored aliases, will be generated on next request
+    public refresh(): void {
+        this.aliases = null;
+        this.allPackageAliases = null;
+    }
+
+    // get all aliases that match the given name, if specified only from 1 package
+    public getAliasesForName(name: string, pkgName?: string): rHelpPanel.Alias[] {
+        const aliases = this.getAliasesForPackage(pkgName);
+        return aliases.filter((v) => v.name === name);
+    }
+
+    // get a list of all aliases
+    public getAllAliases(): rHelpPanel.Alias[] {
+        if(!this.aliases){
+            this.makeAllAliases();
+        }
+        return this.aliases;
+    }
+
+    // get all aliases, grouped by package
+    private getPackageAliases() {
+        if(!this.allPackageAliases){
+            this.readAliases();
+        }
+        return this.allPackageAliases;
+    }
+
+    // get all aliases provided by one package
+    private getAliasesForPackage(pkgName?: string): rHelpPanel.Alias[] | null {
+        if(!pkgName){
+            return this.getAllAliases();
+        }
+        const packageAliases = this.getPackageAliases();
+        if(pkgName in packageAliases){
+            const al = packageAliases[pkgName].aliases;
+            if(al){
+                const ret: rHelpPanel.Alias[] = [];
+                for(const fncName in al){
+                    ret.push({
+                        name: fncName,
+                        alias: al[fncName],
+                        package: pkgName
+                    });
+                }
+                return ret;
+            }
+        }
+        return null;
+    }
+
+    // converts aliases grouped by package to a flat list of aliases
+    private makeAllAliases(): void {
+        if(!this.allPackageAliases){
+            this.readAliases();
+        }
+        const ret: rHelpPanel.Alias[] = [];
+        for(const pkg in this.allPackageAliases){
+            const pkgName = this.allPackageAliases[pkg].package || pkg;
+            const al = this.allPackageAliases[pkg].aliases;
+            if(al){
+                for(const fncName in al){
+                    ret.push({
+                        name: fncName,
+                        alias: al[fncName],
+                        package: pkgName
+                    });
+                }
+            }
+        }
+        this.aliases = ret;
+    }
+
+    // call R script `getAliases.R` and parse the output
+    private readAliases(): void {
+		const lim = '---vsc---'; // must match the lim used in R!
+		const re = new RegExp(`^.*?${lim}(.*)${lim}.*$`, 'ms');
+        const cmd = `${this.rPath} --silent --no-save --no-restore --slave -f ${this.rScriptFile}`;
```

Comment:
I'm curious about how robust it is as many Windows users may have spaces in the absolute paths of both `rPath` and `rScriptFile`?

Reply:
I added quotes for the rScriptFile, the rPath is already quoted [here](https://github.com/ManuelHentschel/vscode-R/blob/bc7714ae2cd002a900b5f0ddb14f096c4a3665ce/src/extension.ts#L59:L61)